src/shared/components/query/quickSearch/QuickSearch.tsx

```diff
@@ -83,76 +86,122 @@ export default class QuickSearch extends React.Component {
-                    type: OptionType.STUDY_COUNT
-                }
-                const geneOptions: any = response[2].body.map(this.geneToOption);
-                const geneCount = {
-                    value: parseInt(response[3].headers["total-count"]) - geneOptions.length,
-                    type: OptionType.GENE_COUNT
-                }
-                const patientOptions: any = response[4].body.map(this.patientToOption);
-                const patientCount = {
-                    value: parseInt(response[5].headers["total-count"]) - patientOptions.length,
-                    type: OptionType.PATIENT_COUNT
-                }
+    // tslint:disable-next-line:member-ordering
+    readonly geneStudyQueryVirtualStudy = remoteData(async () => {
+        const virtualStudyId = AppConfig.serverConfig.skin_quick_search_gene_query_session_id;
 
-                if (((geneOptions.length + patientOptions.length) < (2 * DEFAULT_PAGE_SIZE)) && studyCount.value > 0) {
-                    const spillover = (2 * DEFAULT_PAGE_SIZE) - (geneOptions.length + patientOptions.length);
-                    await client.getAllStudiesUsingGETWithHttpInfo({keyword: input,
-                        pageSize: spillover + DEFAULT_PAGE_SIZE + (SHOW_MORE_SIZE * this.studyPageMultiplier)})
-                        .then((response: any) => {
-                            studyOptions = response.body.map(this.studyToOption);
-                        });
-                    studyCount.value = studyCount.value - spillover;
-                }
+        if (ServerConfigHelpers.sessionServiceIsEnabled() && virtualStudyId) {
+            try {
+                const study = await sessionServiceClient.getVirtualStudy(virtualStudyId);
+                return study;
+            } catch (ex) {
+                return null;
+            }
+        } else {
+            return null;
+        }
+    }, null);
+
+    // tslint:disable-next-line:member-ordering
+    private geneStudyQuery = remoteData<GeneStudyQuery>({
+        await: () => [
+            this.geneStudyQueryVirtualStudy
+        ],
+        invoke: ()=>{
+            if (this.geneStudyQueryVirtualStudy.result) {
+                return Promise.resolve({
+                    type: GeneStudyQueryType.SESSION,
+                    query: this.geneStudyQueryVirtualStudy.result.id,
+                    name: this.geneStudyQueryVirtualStudy.result.data.name,
+                });
+            } else {
+                return Promise.resolve({
+                    type: GeneStudyQueryType.STUDY_LIST,
+                    query: AppConfig.serverConfig.skin_quick_search_gene_query_cancer_study_list,
+                    name: AppConfig.serverConfig.skin_quick_search_gene_query_cancer_study_list_name,
+                });
+            }
+        }
+    });
 
-                let options = [];
+    // tslint:disable-next-line:member-ordering
+    private options = remoteData<any[]>({
```

Comment:
this looks like a huge change, but all it does is add an await for geneStudyQuery, rest is indentation